File: 2121-find-if-path-exists-in-graph/find-if-path-exists-in-graph.py
class Solution:
    def validPath(self, n: int, edges: List[List[int]], source: int, destination: int) -> bool:
        graph = [[] for _ in range(n)]
        
        for e in edges:
            graph[e[0]].append(e[1])
            graph[e[1]].append(e[0])
        
        visited = [False] * n
        visited[source] = True
        queue = deque([source])
        
        while queue:
            temp = queue.popleft()
            
            if temp == destination:
                return True
            
            for connect in graph[temp]:
                if not visited[connect]:
                    visited[connect] = True
                    queue.append(connect)
        
        return False

File: 2121-find-if-path-exists-in-graph/find-if-path-exists-in-graph.cpp
class Solution {
public:
    bool validPath(int n, vector<vector<int>>& edges, int source, int destination) {
        vector<vector<int>> graph(n, vector<int>());
        
        for (const auto& edge : edges) {
            graph[edge[0]].push_back(edge[1]);
            graph[edge[1]].push_back(edge[0]);
        }
        
        vector<bool> visit(n, false);
        visit[source] = true;
        queue<int> q;
        q.push(source);
        
        while (!q.empty()) {
            int temp = q.front();
            q.pop();
            
            if (temp == destination)
                return true;
            
            for (int connect : graph[temp]) {
                if (!visit[connect]) {
                    visit[connect] = true;
                    q.push(connect);
                }
            }
        }
        
        return false;
    }
};